### PR TITLE
MediaService: Add Unit tests

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
MediaService

### Description
This PR adds unit tests, documentation and a quality of life fix for the e2e test.

### Steps

- [x] Added / updated tests
- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: src/media/media.service.ts [hunk 1]
```diff
@@ -59,7 +59,16 @@ export class MediaService {
     return allowedTypes.includes(mimeType);
   }
 
-  public async saveFile(
+  /**
+   * @async
+   * Save the given Buffer to the configured MediaBackend and create a MediaUploadEntity to track where the file is, who uploaded it and to which note.
```
Comment (InnayTool) [inline src/media/media.service.ts:64]:
```suggestion
   * Save the given buffer to the configured MediaBackend and create a MediaUploadEntity to track where the file is, who uploaded it and to which note.
```

File: src/media/media.service.ts [hunk 2]
```diff
@@ -93,7 +102,16 @@ export class MediaService {
...
+  /**
+   * @async
+   * Try to delete the file specified by the filename with the user specified by the username.
+   * @param {string} filename - the name of the file to delete.
+   * @param {string} username - the username of the user who uploaded this file
+   * @return {string} the url of the saved file
+   * @throws {PermissionError} the user is not permitted to delete this file.
+   * @throws {NotInDBError} - the file entry specified is not in the database
+   */
+  async deleteFile(filename: string, username: string): Promise<void> {
```
Comment (InnayTool) [inline src/media/media.service.ts:114]: I think we should remove the username parameter from this and check in the controller if the user has the permission to do that. Therefore it would be good to have an `isOwner` function in the permission service which has a `mediaUpload` object as parameter. Should we make another PR for that or use #777 ?

**Review by InnayTool — CHANGES_REQUESTED** [review]

File: src/media/media.service.ts [hunk 3]
```diff
@@ -112,7 +130,14 @@ export class MediaService {
     await this.mediaUploadRepository.remove(mediaUpload);
   }
 
-  public async findUploadByFilename(filename: string): Promise<MediaUpload> {
+  /**
+   * @async
+   * Find a file entry by their filename.
+   * @return {string} filename - the name of the file entry to find
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply src/media/media.service.ts:136]: Changed.

File: src/media/media.service.ts [hunk 4]
```diff
@@ -112,7 +130,14 @@ export class MediaService {
     await this.mediaUploadRepository.remove(mediaUpload);
   }
 
-  public async findUploadByFilename(filename: string): Promise<MediaUpload> {
+  /**
+   * @async
+   * Find a file entry by their filename.
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply src/media/media.service.ts:135]: changed

File: src/media/media.service.ts (hunk 1 above)
Comment (DerMolly) [reply src/media/media.service.ts:64]: changed

File: src/media/media.service.ts (hunk 2 above)
Comment (DerMolly) [reply src/media/media.service.ts:114]: that is correct. I'd suggest doing that in a different PR.

Comment (InnayTool) [reply src/media/media.service.ts:114]: ok

**Review by InnayTool — DISMISSED** [review]

**Review by InnayTool — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]
